pkg/machine/qemu/machine.go

```diff
@@ -507,29 +516,87 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 		}
 	}
 
-	// If the qemusocketpath exists and the vm is off/down, we should rm
-	// it before the dial as to avoid a segv
-	if err := v.QMPMonitor.Address.Delete(); err != nil {
+	vlanSocket, err := machineSocket(v.Name, "vlan", "")
+	if err != nil {
 		return err
 	}
-	for i := 0; i < 6; i++ {
-		qemuSocketConn, err = net.Dial("unix", v.QMPMonitor.Address.GetPath())
-		if err == nil {
-			break
+
+	isFdVlanVM := false
+	cmdLine := v.CmdLine
+	for _, c := range v.CmdLine {
+		if c == fdVlanNetdev() {
+			isFdVlanVM = true
 		}
-		time.Sleep(wait)
-		wait++
 	}
-	if err != nil {
-		return err
+
+	cmdLine = propagateHostEnv(cmdLine)
+
+	// Show migration recommendation if machine was created with different settings
+	if isFdVlanVM != useFdVLan() {
+		logrus.Warn("Stored Podman Machine configuration doesn't match current settings")
```

Comment:
It checks that current setting in use is different from the time, when the machine was created. I updated the description (creating separate ones for either branch) and also demoted logging to DEBUG level to reduce noise (as both implementations should have same behavior on platforms, which supports both of them and no need to bother the user unless the troubleshooting happening).

These logs lines are pretty optional in general and can be freely removed. They raise from good intentions, but looks sort of trouble makers on their own, because tinkering with QEMU cmd in the machine config is sort of a feature of Podman.

Reply:
ok so re-examining the overall logic, it seems the design CONTAINERS_USE_SOCKET_VLAN is very much a machine init (newmachine) feature, and really has no role on the start() path. So IMO its probably better to simplify the code path and just remove the extra logic here.